### PR TITLE
refacort: improve code logic to increase robustness

### PR DESCRIPTION
**Pull request by Wxh16144**
backround: https://github.com/un-ts/changesets-gitlab/issues/112#issuecomment-1664178045

### CODE REVIEW

**Comment by Wxh16144** [comment]
[GitLab Predefined variables reference](https://docs.gitlab.com/ee/ci/variables/predefined_variables.html)

**Review by JounQin — CHANGES_REQUESTED** [review]

File: src/index.ts [hunk 1]
```diff
@@ -23,7 +23,13 @@ export const createApi = (gitlabToken?: string) => {
     }
   }
 
-  const token = gitlabToken || process.env.GITLAB_TOKEN
+  const {
+    GITLAB_TOKEN,
+    CI_SERVER_PROTOCOL,
+    CI_SERVER_HOST,
```
[the remark this answers is not in the document]
Comment (Wxh16144) [reply src/index.ts:29]: I'm not sure if I need the port, because when I set it to 80, it throws an error

**Comment by Wxh16144** [comment]
> Hi, @Wxh16144 , thanks for your contribution! Please fix the CI issues first.

I don't speak English very well, but seeing that you are also Chinese, can I reply you in Chinese?
我英文不太好,  看到你也是华人, 我可以用中文回复你吗?

File: src/index.ts (hunk 1 above)
[the remark this answers is not in the document]
Comment (JounQin) [reply src/index.ts:29]: @Wxh16144 当然，用 `CI_SERVER_URL` 也会报错吗？

**Review by JounQin — APPROVED** [review]

**Pull request merged**